control/preimage/src/commands.rs

```diff
@@ -1,44 +1,166 @@
+pub fn upgrade(params: &UpgradeArgs) -> BridgeHubRuntimeCall {
+    let initializer = if params.initializer {
+        Some((
+            params.initializer_params.as_ref().unwrap().clone(),
+            params.initializer_gas.unwrap(),
+        ))
+    } else {
+        None
     };
 
-    let call = BridgeHubRuntimeCall::EthereumSystem(snowbridge_pallet_system::pallet::Call::set_operating_mode { mode });
-    let call = wrap_calls(context, vec![call]).await?;
-    Ok(call)
+    BridgeHubRuntimeCall::EthereumSystem(snowbridge_pallet_system::pallet::Call::upgrade {
+        impl_address: params.logic_address.into_array().into(),
+        impl_code_hash: params.logic_code_hash.0.into(),
+        initializer: initializer.map(|(params, gas)| Initializer {
+            params: params.into(),
+            maximum_required_gas: gas,
+        }),
+    })
 }
 
-pub async fn upgrade(
+pub async fn pricing_parameters(
     context: &Context,
-    logic_address: Address,
-    logic_code_hash: FixedBytes<32>,
-    initializer: Option<(Bytes, u64)>,
-) -> Result<RelayRuntimeCall, Box<dyn std::error::Error>> {
-    let call = BridgeHubRuntimeCall::EthereumSystem(
-        snowbridge_pallet_system::pallet::Call::upgrade {
-            impl_address: H160::from_slice(logic_address.as_slice()),
-            impl_code_hash: H256::from_slice(logic_code_hash.as_slice()),
-            initializer: initializer.map(|(params, gas)| Initializer {
-                params: params.into(),
-                maximum_required_gas: gas,
-            }),
-        }
+    params: &PricingParametersArgs,
+) -> Result<(BridgeHubRuntimeCall, AssetHubRuntimeCall), Box<dyn std::error::Error>> {
+    // BridgeHub parameters
+    let pricing_params: PricingParameters<u128> = PricingParameters {
+        exchange_rate: Static(FixedU128::from_rational(
+            params.exchange_rate_numerator.into(),
+            params.exchange_rate_denominator.into(),
+        )),
+        multiplier: Static(FixedU128::from_rational(
+            params.multiplier_numerator.into(),
+            params.multiplier_denominator.into(),
+        )),
+        fee_per_gas: bridge_hub_rococo_runtime::runtime_types::primitive_types::U256(
+            U256::from(GWEI_UNIT)
+                .checked_mul(U256::from(params.fee_per_gas))
+                .unwrap()
+                .into_limbs(),
+        ),
+        rewards: Rewards {
+            local: params.local_reward.to::<u128>(),
+            remote: bridge_hub_rococo_runtime::runtime_types::primitive_types::U256(
+                params.remote_reward.into_limbs(),
+            ),
+        },
+    };
+
+    let outbound_delivery_fee = calculate_delivery_fee(&pricing_params).await?;
+
+    let total_outbound_fee = outbound_delivery_fee.local + outbound_delivery_fee.remote;
+
+    // Adjust outbound fee up by 10% as a buffer
+    let total_outbound_fee_adjusted = total_outbound_fee.saturating_add(total_outbound_fee / 10);
```

Comment:
Yes, I think its enough, let me explain:

The outbound fee needs to cover both:
1. Local costs of processing of BridgeHub,
2. Cost of delivery to Ethereum, which is substantially more than (1)

Component (1) can fluctuate in ways that are outside of our control, for example, updated weights or changes to substrate that result in fee changes.

Component (2) is under our control, as long as we make sure to update the fee on AssetHub when changing the pricing parameters on BridgeHub.

This is why I think we can go with a reduced buffer, since component (1) is much much less than component (2)